src/context.rs

```diff
@@ -926,6 +961,24 @@ impl Context {
         self.context.opaque_struct_type(name)
     }
 
+    /// Get type by its name, if it exists.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    ///	let name = "opaque";
+    ///	let opaque = context.opaque_struct_type(name);
+    ///
+    /// let got = context.get_type(name);
+    ///	assert_eq!(got, Some(AnyTypeEnum::from(opaque)));
+    ///
+    /// assert_eq!(context.get_type("non-existent"), None);
+    /// ```
+    #[inline]
+    pub fn get_type<'ctx>(&self, name: &str) -> Option<AnyTypeEnum<'ctx>> {
```

Comment:
We should add this to ContextRef as well

Reply:
Done in 48af9c33817a881eaa797f954bbf02b2f258c9ba